src/Finder.elm

```diff
@@ -130,31 +140,38 @@ update env msg model =
         NoOp ->
             ( model, Cmd.none, Remain )
 
-        UpdateQuery query ->
+        UpdateInput input ->
             let
                 isSequenceShortcutInput =
-                    String.contains ";" query
+                    String.contains ";" input
             in
-            if String.isEmpty query then
-                ( { model | query = query, search = NotAsked }, Cmd.none, Remain )
+            if String.isEmpty input then
+                ( { model | input = input, search = NotAsked }, Cmd.none, Remain )
+
+            else if String.length input > 1 && not isSequenceShortcutInput then
+                ( { model | input = input }, Util.delayMsg debounceDelay (PerformQuery input), Remain )
+
+            else if not isSequenceShortcutInput then
+                ( { model | input = input }, Cmd.none, Remain )
 
-            else if String.length query > 1 && not isSequenceShortcutInput then
+            else
+                ( model, Cmd.none, Remain )
+
+        PerformQuery query ->
+            if query == model.input then
```

Comment:
Instead of doing anything fancy with comparing times etc, we just wait 300ms and if the query hasn't changed, we perform the request.